core-ui/src/components/Predefined/Create/Subscriptions/Subscriptions.create.js

```diff
@@ -23,58 +23,57 @@ const SubscriptionsCreate = ({
   onChange,
   formElementRef,
   namespace,
-  resource: initialEventSubscription,
+  resource: initialSubscription,
   resourceUrl,
-  serviceName,
+  serviceName = '',
   setCustomValid,
 }) => {
   const { t } = useTranslation();
-  const { data: configMap } = useGet(
-    '/api/v1/namespaces/kyma-system/configmaps/eventing',
-  );
-
-  let eventTypePrefix =
-    configMap?.data?.eventTypePrefix || DEFAULT_EVENT_TYPE_PREFIX;
-  eventTypePrefix = eventTypePrefix.endsWith('.')
-    ? eventTypePrefix
-    : eventTypePrefix + '.';
 
-  const [eventSubscription, setEventSubscription] = useState(
-    cloneDeep(initialEventSubscription) ||
-      createEventSubscriptionTemplate(namespace, eventTypePrefix),
+  const [subscription, setSubscription] = useState(
+    cloneDeep(initialSubscription) ||
+      createSubscriptionTemplate(
+        namespace,
+        DEFAULT_EVENT_TYPE_PREFIX,
+        serviceName,
+      ),
   );
 
   const firstEventType = jp.value(
-    eventSubscription,
+    subscription,
     '$.spec.filter.filters[0].eventType.value',
   );
-  const firstEventTypeValues = spreadEventType(firstEventType, eventTypePrefix);
+
+  const firstEventTypeValues = spreadEventType(
+    firstEventType,
+    DEFAULT_EVENT_TYPE_PREFIX,
+  );
 
   const notification = useNotification();
 
   useEffect(() => {
```

Comment:
do we need this useEffect? wouldn't `setValue` that we have in line 154 be enough?